Tools/LyTestTools/ly_test_tools/cli/codeowners_hint.py

```diff
@@ -0,0 +1,232 @@
+                    return owned_path, aliases
+                # else invalid entry syntax, ignore
+
+    return "", ""  # no match found
+
+
+def _codeowners_path_matches(target_path: pathlib.PurePosixPath, owned_path: str) -> bool:
+    """
+    :param target_path: PurePosixPath to match against, which starts from the root of the repo
+    :param owned_path: path identifier found in a GitHub CODEOWNERS file (relative to root, may contain wildcards)
+    :return: True when target_path is contained by the rules of owned_path
+    """
+    matched = False
+    if '*' in owned_path or '?' in owned_path:  # wildcards require glob matching
+        if owned_path.startswith("*"):  # special simple case for global wildcards
+            matched = target_path.match(owned_path)
+        elif owned_path.startswith("/"):  # ownership of specific directory: glob A against B
+            matched = target_path.match(owned_path[1:])
+        else:  # ownership of all relative directories: find non-wildcard portions of B in A, glob the remainders
+            asterisk = owned_path.find("*")
+            question = owned_path.find("?")
+            if asterisk > -1 and question > -1:
+                first_wildcard_index = min(asterisk, question)
+            else:  # avoid not-found index
+                first_wildcard_index = max(asterisk, question)
+            separator_indices = [index.start() for index in re.finditer(pattern="/", string=owned_path)]
+            pre_wildcard_separator_index = 0
+            for s_index in separator_indices:
+                if s_index < first_wildcard_index:
+                    pre_wildcard_separator_index = s_index
+                else:  # remainder are all greater
+                    break
+
+            # separate non-wildcard-containing path from remainder
+            pre_wildcard_owned = owned_path[:pre_wildcard_separator_index]
+            wildcard_with_remainder_owned = owned_path[pre_wildcard_separator_index+1:]
+
+            # find substrings of initial portion of B within A
+            target_str = str(target_path)
+            pre_wildcard_target_end_indices = [index.end() for index in
+                                               re.finditer(pattern=pre_wildcard_owned, string=target_str)]
+            # glob remainders of A against remainder of B
+            for target_index in pre_wildcard_target_end_indices:  # may be multiple substring matches within target
+                target_remainder = target_str[target_index:]
+                if pathlib.PurePosixPath(target_remainder).match(wildcard_with_remainder_owned):
+                    matched = True
+                    break  # exit early on success
+    else:  # simple path matching
+        if owned_path.startswith("/"):  # ownership of specific directory: verify if A exists inside B
+            try:  # is_relative_to introduced in Python 3.9
+                target_path.relative_to(owned_path)
+                matched = True
```

Comment:
If you wait a couple of days, @spham-amzn is getting Python 3.10 to be downloaded as the next python.

Reply:
Given the minor/secondary benefit of this change, I see little reason to not wait for the new python to ship and simplify this code.